uma/nonce_cache.py

```diff
@@ -0,0 +1,52 @@
+# Copyright ©, 2022-present, Lightspark Group, Inc. - All Rights Reserved
+
+from abc import ABC, abstractmethod
+from typing import Dict
+from uma.exceptions import InvalidNonceException
+
+
+class INonceCache(ABC):
+    @abstractmethod
+    def check_and_save_nonce(self, nonce: str, timestamp: int) -> None:
```

Comment:
Would it make sense to take a datetime here instead of an int?

Reply:
yeah, that sounds good -- the request/response timestamps are inconsistent (some use datetime, some use int), maybe we should also unify them to all use datetime?

Reply:
Hmm yeah might be worthwhile. My fault!

Reply:
sg! will do that in v1.0